src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -848,11 +832,161 @@ export class BrowsingContextImpl {
     }
   }
 
+  async #parseRect(clip?: BrowsingContext.ClipRectangle) {
+    if (!clip) {
+      return;
+    }
+    switch (clip.type) {
+      case 'viewport':
+        return {x: clip.x, y: clip.y, width: clip.width, height: clip.height};
+      case 'element': {
+        if (clip.scrollIntoView) {
+          throw new UnsupportedOperationException(
+            `'scrollIntoView' is currently not supported`
+          );
+        }
+        const sandbox = await this.getOrCreateSandbox(undefined);
+        const result = await sandbox.callFunction(
+          String((element: unknown) => {
+            return element instanceof Element;
+          }),
+          {type: 'undefined'},
+          [clip.element],
+          false,
+          Script.ResultOwnership.None,
+          {}
+        );
+        if (result.type === 'exception') {
+          throw new NoSuchElementException(
+            `Element ${clip.element.sharedId} was not found`
+          );
+        }
+        assert(result.result.type === 'boolean');
+        if (!result.result.value) {
+          throw new NoSuchElementException(
+            `Node ${clip.element.sharedId} is not an Element`
+          );
+        }
+        {
+          const result = await sandbox.callFunction(
+            String((element: Element) => {
+              const rect = element.getBoundingClientRect();
+              return {
+                x: rect.x,
+                y: rect.y,
+                height: rect.height,
+                width: rect.width,
+              };
+            }),
+            {type: 'undefined'},
+            [clip.element],
+            false,
+            Script.ResultOwnership.None,
+            {}
+          );
+          assert(result.type === 'success');
+          const rect = deserializeDOMRect(result.result);
+          if (!rect) {
+            throw new UnableToCaptureScreenException(
+              `Could not get bounding box for Element ${clip.element.sharedId}`
+            );
+          }
+          return rect;
+        }
+      }
+    }
+  }
+
   async close(): Promise<void> {
     await this.#cdpTarget.cdpClient.sendCommand('Page.close');
   }
 }
 
+type Rect = {
```

Comment:
Why?

Use https://chromedevtools.github.io/devtools-protocol/tot/DOM/#type-Rect (`DOM.Rect`) instead, so that we do not reinvent the wheel.